internal/venice-client-common/src/main/java/com/linkedin/venice/utils/AvroSupersetSchemaUtils.java

```diff
@@ -34,56 +34,56 @@ public static boolean isSupersetSchema(Schema s1, Schema s2) {
    * Generate super-set schema of two Schemas. If we have {A,B,C} and {A,B,D} it will generate {A,B,C,D}, where
    * C/D could be nested record change as well eg, array/map of records, or record of records.
    * Prerequisite: The top-level schema are of type RECORD only and each field have default values. ie they are compatible
-   * schemas and the generated schema will pick the default value from s1.
-   * @param existingSchema schema existing in the repo
-   * @param newSchema schema to be added.
+   * schemas and the generated schema will pick the default value from new value schema.
+   * @param newSchema schema existing in the repo
+   * @param oldSchema schema to be added.
    * @return super-set schema of existingSchema abd newSchema
    */
-  public static Schema generateSuperSetSchema(Schema existingSchema, Schema newSchema) {
-    if (existingSchema.getType() != newSchema.getType()) {
+  public static Schema generateSuperSetSchema(Schema newSchema, Schema oldSchema) {
```

Comment:
I feel not changing the function definition should make the diff a bit concise. what do you think?

Reply:
Do you mean reverse the order back? Let me see if I can achieve that.

Reply:
Fixed. Could you check again?